src/ETISS.cpp

```diff
@@ -904,6 +893,48 @@ static class helper_class_etiss_1
 
 bool etiss_shutdownOk = false;
 
+void etiss::initialize_virtualstruct(std::shared_ptr<etiss::CPUCore> cpu_core)
+{
+    auto mount_successful = etiss::VirtualStruct::root()->mountStruct(cpu_core->getName(), cpu_core->getStruct());
+
+    if(!mount_successful)
+    {
+        etiss::log(etiss::FATALERROR, std::string("Failed to mount ") + cpu_core->getName() + std::string("'s VirtualStruct, but failed: etiss::CPUCore not created!"));
+    }
+    else
+    {
+        etiss::log(etiss::VERBOSE, std::string("Mounted ") + cpu_core->getName() + std::string("'s VirtualStruct to root VirtualStruct"));
+
+        // load fault files
+        std::string faults = cfg().get<std::string>("faults.xml", "");
+        if (!faults.empty())
+        {
+            std::list<std::string> ffs = etiss::split(faults, ';');
+            for (auto ff : ffs)
```

Comment:
The default pattern for this should be "const auto&". If you need mutablity, drop const and if you need a copy, drop the reference.

Reply:
agreed (copied from old code base). Fixed with https://github.com/tum-ei-eda/etiss/pull/100/commits/3087caeffd1073650c85b86f29fb1228f4cce22f